### ISSUE TITLE
Question - opening files 

### ISSUE THREAD

**Issue by KasselMarc** [ahrm/sioyek#1248]
I always open files from a specific folder. Can I somehow configure Sioyek command so that I can open files from a specific directory that is also searchable? I'm running it on MacOS.

I know that one can search through all files one has previously interacted with, but I want a list that is somewhat static and only contains files from a specific folder.

Thanks for this great pdf viewer, the best I've ever used!

**Comment by ahrm** [comment]
There is an `open_document_embedded_from_current_path` command that does that.

**Comment by KasselMarc** [comment]
> There is an `open_document_embedded_from_current_path` command that does that.

Yes, I have tried this but it doesn't work (i.e. even if I enter the entire file name and hit enter, it won't open).

**Comment by KasselMarc** [comment]
> There is an `open_document_embedded_from_current_path` command that does that.

Sorry for having bothered you, for some reason I didn't give the program access to some folders and that's why it didn't work when I initially tried it. Thanks a lot!

**Comment by KasselMarc** [comment]
The command you mentioned provided the correct functionality, but I would like to bind a key like the following:
`open_document_from_directory [the directory] <key>`. Is this already possible?

**Comment by ahrm** [comment]
This is not possible, however you can achieve this using a hack like this: You can set a capital letter mark (say `M`) in a document in the target folder and then add the following keybind in `keys_user.config`:
```
goto_mark(M);open_document_embedded_from_current_path <keybind>
```

**Comment by KasselMarc** [comment]
> This is not possible, however you can achieve this using a hack like this: You can set a capital letter mark (say `M`) in a document in the target folder and then add the following keybind in `keys_user.config`:
> 
> ```
> goto_mark(M);open_document_embedded_from_current_path <keybind>
> ```

Okay thanks, one last question: Is there some way to interact with sioyek using the terminal (or perhaps the python wrapper you wrote)? I want to build a raycast extension which basically executes `open_document_from_directory` on the current selected sioyek window.

**Comment by ahrm** [comment]
You can run
```
sioyek --execute-command open_document_embedded_from_current_path
```

**Comment by ahrm** [comment]
It is actually:
```
sioyek --execute-command open_document_embedded_from_current_path --window-id 0
```

**Comment by KasselMarc** [comment]
> It is actually:
> 
> ```
> sioyek --execute-command open_document_embedded_from_current_path --window-id 0
> ```

That's great! Where can I find the documentation of this (apparently I'm just blind)?

**Comment by ahrm** [comment]
No.

**Comment by KasselMarc** [comment]
> No.

Ok, that's what I wanted to know. Thanks!

**Comment by ahrm** [comment]
Doesn't raycast support running multiple commands? You could do this using two commands by first opening the file:
```
sioyek --window-id 0 /path/to/file
```
and then running `open_document_from_current_path` command.

**Comment by KasselMarc** [comment]
When I run `sioyek --execute-command open_document_embedded_from_current_path --window-id 0` I get `Unknown option 'window-id'`.

**Comment by KasselMarc** [comment]
> Doesn't raycast support running multiple commands? You could do this using two commands by first opening the file:
> 
> ```
> sioyek --window-id 0 /path/to/file
> ```
> 
> and then running `open_document_from_current_path` command.

Sorry, I misunderstood your last response. Yes, this should work fine! I'm currently just struggling with the `window-id` option.

**Comment by KasselMarc** [comment]
Is the option `--window-id` already available in the current v2.0.0 or do I have to install the Pre-release version? In any way, having just two windows open one can work around this.

**Comment by ahrm** [comment]
You need the pre-release version.
